go18_test.go

```diff
@@ -155,6 +156,32 @@ func TestContextCancelQuery(t *testing.T) {
 	}
 }
 
+// TestIssue617 tests that a failed query in QueryContext doesn't lead to a
+// goroutine leak.
+func TestIssue617(t *testing.T) {
+	db := openTestConn(t)
+	defer db.Close()
+
+	const N = 10
+
+	numGoroutineStart := runtime.NumGoroutine()
+	for i := 0; i < N; i++ {
+		query := `SELECT * FROM DOESNOTEXIST`
+		ctx, _ := context.WithTimeout(context.Background(), time.Second)
```

Comment:
heh, this also leaks a goroutine. I understand that what you want here is a cancellable context, so please write this as:

```
func() {
  ctx, cancel := context.WithCancel(context.Background())
  defer cancel()
  rows, err := db.QueryContext(ctx, `SELECT * FROM DOESNOTEXIST`)
  ....
}()